### ISSUE TITLE
_get_repo_root() in `noxfile.py` can fail when `.git` does not exist at project root

### ISSUE THREAD

**Issue by anguillanneuf** [googleapis/synthtool#792]
Using `.git` to search for the project root only works in repos that have a `.git` folder present. 

https://github.com/googleapis/synthtool/blob/487eba79f8260e34205d8ceb1ebcc65685085e19/synthtool/gcp/templates/python_samples/noxfile.py.j2#L195-L205

But many of our repos don't have a `.git` folder at the project root level, including:
- https://github.com/googleapis/python-bigquery
- https://github.com/googleapis/python-pubsub

Using `.gitignore` is probably better.

TL;DR - I came across this problem when using Cloud Build for `python-pubsublite` samples testing. Cloud Build copies all the files in a repo to a folder named `workspace/`. When I `cd` into `samples/snippets/` and do `pip install -e ..`, I keep getting an error that says "Unable to detect repository root".

**Comment by busunkim96** [comment]
Does Cloud Build copy the repo instead of cloning the repo? `.git` is a folder that git itself uses. If you clone a repo it is always in the root. (https://stackoverflow.com/questions/29217859/what-is-the-git-folder) I think that's why it was chosen for the repo root detection code. 

I think we could try looking for multiple things. `.git` and something else that only is in the root. `setup.py` maybe? Technically a repo can have `.gitignore`s in nested directories.

**Comment by anguillanneuf** [comment]
Indeed they don't include `.git`: https://github.com/GoogleCloudPlatform/cloud-builders/issues/236
